lib/protoflow-blocks/src/blocks/flow/sort.rs

```diff
@@ -0,0 +1,152 @@
+// This is free and unencumbered software released into the public domain.
+
+use crate::{StdioConfig, StdioError, StdioSystem, System};
+use protoflow_core::{
+    prelude::Vec, types::Any, Block, BlockResult, BlockRuntime, InputPort, Message, OutputPort,
+};
+use protoflow_derive::Block;
+
+/// A block that simply stores all messages it receives.
+///
+/// # Block Diagram
+
+///
+/// # Sequence Diagram
+
+///
+/// # Examples
+///
+/// ## Using the block in a system
+///
+/// ```rust
+/// # use protoflow_blocks::*;
+/// # fn main() {
+/// System::build(|s| {
+///     let stdin = s.read_stdin();
+///     let sort = s.sort();
+///     s.connect(&stdin.output, &sort.input);
+/// });
+/// # }
+/// ```
+///
+/// ## Running the block via the CLI
+///
+/// ```console
+/// $ protoflow execute Sort
+/// ```
+///
+#[derive(Block, Clone)]
+pub struct Sort<T: Message = Any> {
+    /// The input message stream.
+    #[input]
+    pub input: InputPort<T>,
+
+    #[input]
+    pub stop: InputPort<T>,
+
+    /// The output message stream.
+    #[output]
+    pub output: OutputPort<T>,
+
+    /// The internal state storing the messages received.
+    #[state]
+    messages: Vec<T>,
+}
+
+impl<T: Message> Sort<T> {
+    pub fn new(input: InputPort<T>, stop: InputPort<T>, output: OutputPort<T>) -> Self {
+        Self {
+            input,
+            stop,
```

Comment:
```suggestion
    pub fn new(input: InputPort<T>, output: OutputPort<T>) -> Self {
        Self {
            input,
```

Reply:
I use it to identify the stop message, which triggers sending the buffered messages to the output port, what would be the alternative solution in this case?

Reply:
I think the more natural behaviour would be that `Sort` block consumes its input to the end, sorts, and then sends on output port.
Assume that you're sorting integers, what is the "signal" value? Max int? Min int? 0? There's no reason why those can't also be valid data.

In any case you're mixing a signaling method inside the stream of data and at the very least it should not be comparing (commented implementation currently has `if message == x`) each of the data messages against the signal message.

Reply:
you're right about "I think the more natural behaviour would be that Sort block consumes its input to the end, sorts, and then sends on output port.". that would be better